assets/javascripts/kitten/components/buttons/tag-button/index.js

```diff
@@ -48,132 +49,150 @@ const StyledTagButton = styled(
+  }
 
-  &:hover,
-  &:focus {
-    color: ${COLORS.background1};
-    border-color: ${COLORS.primary2};
-    background-color: ${COLORS.primary2};
+  &.k-Buttons--TagButton__huge {
+    min-height: ${pxToRem(60)};
+    min-width: ${pxToRem(80)};
+    padding: 0 ${pxToRem(30)};
+    font-size: ${stepToRem(-1)};
+    @media screen and (-ms-high-contrast: active), (-ms-high-contrast: none) {
+      height: ${pxToRem(60)};
+      width: ${pxToRem(80)};
+    }
   }
 
-  &:active {
-    color: ${COLORS.primary2};
-    border-color: ${COLORS.primary2};
+  &.k-Buttons--TagButton__helium,
+  &.k-Buttons--TagButton__hydrogen {
+    color: ${COLORS.font1};
+    border-color: ${COLORS.line1};
     background-color: ${COLORS.background1};
+  
+    &:hover,
+    &:focus {
+      color: ${COLORS.background1};
+      border-color: ${COLORS.primary2};
+      background-color: ${COLORS.primary2};
+    }
+  
+    &:active {
+      color: ${COLORS.primary2};
+      border-color: ${COLORS.primary2};
+      background-color: ${COLORS.background1};
+    }
   }
-`
 
-export class TagButton extends PureComponent {
-  static propTypes = {
-    modifier: PropTypes.string,
-    tiny: PropTypes.bool,
-    big: PropTypes.bool,
-    icon: PropTypes.bool,
-    selected: PropTypes.bool,
-  }
+  &.k-Buttons--TagButton__carbon {
+    color: ${COLORS.font1};
+    border-color: ${COLORS.background1};
+    background-color: ${COLORS.background1};
+
+    &:hover,
+    &:focus {
+      color: ${COLORS.primary1};
+      border-color: ${COLORS.background1};
+      background-color: ${COLORS.background1};
+    }
 
-  static defaultProps = {
-    children: 'Tag',
-    modifier: 'hydrogen',
-    icon: false,
-    tiny: false,
-    big: false,
-    selected: false,
+    &:active {
+      color: ${COLORS.primary1};
+      border-color: ${COLORS.background1};
+      background-color: ${COLORS.background1};
+    }
   }
-  render() {
-    const {
-      children,
-      modifier,
-      selected,
-      tiny,
-      big,
-      icon,
-      ...others
-    } = this.props
-
-    return (
-      <StyledTagButton
-        modifier={modifier}
-        selected={selected}
-        tiny={tiny}
-        big={big}
-        icon={icon}
-        {...others}
-      >
-        {children}
-      </StyledTagButton>
-    )
+
+  &.k-Buttons--TagButton__selected {
+    color: ${COLORS.background1};
+    border-color: ${COLORS.primary1};
+    background-color: ${COLORS.primary1};
   }
+`
+
+export const TagButton = ({
+  children,
+  modifier,
+  selected,
+  tiny,
+  big,
+  huge,
+  icon,
+  ...others
+}) => {
+
+  return (
+    <StyledTagButton
+      className={classNames(
+        'k-Buttons--TagButton',
+        `k-Buttons--TagButton__${modifier}`,
+        {
+          'k-Buttons--TagButton__tiny': tiny,
+          'k-Buttons--TagButton__big': big,
+          'k-Buttons--TagButton__huge': huge,
+          'k-Buttons--TagButton__selected': selected,
+          'k-Buttons--tagButton__icon': icon,
+        }
```

Comment:
Je te recommande de suivre le BEM, ça ferait `k-Buttons__tagButton`, `k-Buttons__tagButton--modifier`, `k-Buttons__tagButton--selected` …

```suggestion
        'k-Buttons__tagButton',
        `k-Buttons__tagButton--${modifier}`,
        {
          'k-Buttons__tagButton--tiny': tiny,
          'k-Buttons__tagButton--big': big,
          'k-Buttons__tagButton--huge': huge,
          'k-Buttons__tagButton--selected': selected,
          'k-Buttons__tagButton--icon': icon,
        }
```